### PR TITLE
update SDK Registry plugin to v0.7.0

### PR DESCRIPTION
**Pull request by LukasPaczos**
### Description

This upgrade makes the SDK Registry PR creation automatic.

The SDK Versions plugin upgrade is needed for compatibility.

### CODE REVIEW

**Review by Zayankovsky — APPROVED** [review]

**Comment by LukasPaczos** [comment]
@Zayankovsky I refactored the CI script a little bit, could you give this another pass?

File: Makefile [hunk 1]
```diff
@@ -131,27 +111,32 @@ core-update-api: assemble-core-release
...
 ui-unit-tests-jacoco:
-	$(call run-gradle-tasks,$(UI_MODULES),jacocoTestReport) \
-	&& ./gradlew libnavui-dropin:jacocoTestReport
+	$(call run-gradle-tasks,$(UI_MODULES),jacocoTestReport)
+
+.PHONY: publish-local
+core-publish-local:
+	./gradlew publishToMavenLocal
+
+.PHONY: upload-to-sdk-registry
```
Comment (Zayankovsky) [inline Makefile:132]: Not sure I understand all changes correctly, but looks like previously this task didn't consider `libnavui-dropin` module, but now it does. Is it expected?

Comment (LukasPaczos) [reply Makefile:132]: `libnavui-dropin` does not apply the `publish.gradle` script so the publishing tasks are not available for that module.

**Comment by LukasPaczos** [comment]
Ready for another round @Zayankovsky.

File: .circleci/config.yml [hunk 2]
```diff
@@ -390,15 +390,13 @@ commands:
...
-    parameters:
-      artifact-type:
-        description: core or ui
-        type: string
     steps:
       - deploy:
           name: Upload Navigation SDK to SDK Registry
           command: |
-            make << parameters.artifact-type >>-upload-to-sdk-registry
+            export GITHUB_TOKEN=$(./mbx-ci github writer token)
```
Comment (Zayankovsky) [inline .circleci/config.yml:397]:
```suggestion
            export GITHUB_TOKEN=$(./mbx-ci github writer public token)
```
Or I can update it in #5233 myself.

Comment (LukasPaczos) [reply .circleci/config.yml:397]: Happy to add it here.

**Comment by LukasPaczos** [comment]
Interestingly, trying to use the new SDK registry plugin uncovers the incompatibility with `androidx.lifecycle:lifecycle-runtime-ktx:2.4.0` which requires Android SDK version 31. Refs https://github.com/mapbox/mapbox-navigation-android/pull/5102.
```
   > A failure occurred while executing com.android.build.gradle.internal.tasks.CheckAarMetadataWorkAction
      > The minCompileSdk (31) specified in a
        dependency's AAR metadata (META-INF/com/android/build/gradle/aar-metadata.properties)
        is greater than this module's compileSdkVersion (android-30).
        Dependency: androidx.lifecycle:lifecycle-runtime-ktx:2.4.0.
        AAR metadata file: /root/.gradle/caches/transforms-2/files-2.1/7aa2e79458120658aaa1de03b1087fab/jetified-lifecycle-runtime-ktx-2.4.0/META-INF/com/android/build/gradle/aar-metadata.properties.
```

Wondering how does this relates to the plugin upgrade and only shows up now. Changing the version back to 0.4.0 removes the problem.

**Comment by LukasPaczos** [comment]
This looks to be directly related to the AGP version used, our project is on `4.0.1` and the new SDK Registry plugin is on `4.1.3` so we get it transitively. If I just bump our AGP version to `4.1.3` I can reproduce the issue as well. Somewhere between these versions, the AAR metadata started to be respected.

**Comment by LukasPaczos** [comment]
I think we can consider this blocked until https://github.com/mapbox/mapbox-navigation-android/pull/5102 is merged.

**Comment by Zayankovsky** [comment]
> I think we can consider this blocked until #5102 is merged.

Can't we just bump `compileSdkVersion`?

File: Makefile [hunk 3]
```diff
@@ -17,6 +17,7 @@ libnavui-resources \
 libnavui-voice \
 libnavigation-android \
 libnavui-speedlimit \
+libnavui-dropin \
```
[the remark this answers is not in the document]
Comment (LukasPaczos) [reply Makefile:20]: @RingerJK the drop-in module doesn't apply the publishing script so publishing tasks won't be available.

**Comment by LukasPaczos** [comment]
> Can't we just bump compileSdkVersion to 31? We don't have to bump targetSdkVersion with it.

I didn't want to mix it up with other PRs that do the same, but I guess it doesn't really matter, we can do it here since the other PRs are blocked.

**Review by RingerJK — APPROVED** [review]

**Comment by LukasPaczos** [comment]
Due to other infrastructural issues, removing the Android SDK and versions plugin bumps in favor of rebasing on top of https://github.com/mapbox/mapbox-navigation-android/pull/5192 when it lands.

**Pull request merged**